### ISSUE TITLE
5/31/17 ER Issues

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#372]
In GitLab by @kdjstudios on May 31, 2017, 14:47

Hello team,

Today there have been three main issues reported that are stopping sites from completing their billing:

* Can not create a new account. Will be presented with an error. This effects all users, including Tim's and mine. #370 
* GM's can not edit draft invoices when processing the billing cycle. #371
* GM's can not create or edit manual draft invoices. 

There was in addition to these another issue that was reported yesterday:

*  Billings has 160 missing accounts. I believe Tim was able to do a manual adjustment in the database to correct this. But we are currently waiting on the replication server before we test. #361

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 1, 2017, 02:34

@kdjstudios 

* **Can not create a new account. Will be presented with an error. This affects all users, including Tim's and mine. #370** 
  * We can now create a new account. 
  * This error was independent of user's access ability.
  * This happened due to a bug fix we did for #341 

* GM's can not edit draft invoices when processing the billing cycle. #371   
##### Following is the Manual Invoice actions and access chart  
    * Create Manual Invoice: **_o4, o5, o6, admin_** (manager and admin in case of hosted)
    * Edit Draft: **_o4, o5, o6, admin_** (manager and admin in case of hosted)
    * Lock/ finalize Draft Invoice: **_o4, o5, o6, admin_** (manager and admin in case of hosted)
    * Delete Draft Invoice: **_ o5, o6, admin_** (manager and admin in case of hosted)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Jun 1, 2017, 08:44

@tim.traylor 

Should GM's (o4) have the ability to delete a draft? They will be the primary users to use this feature I believe.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Jun 2, 2017, 05:23

@kdjstudios 

As per the current functionality **o4** does not have the ability to delete manually drafted invoices, just let us know if we have to keep it this way.

Thanks
